pkg/providers/vsphere/config/template-md.yaml

```diff
@@ -72,19 +72,19 @@ spec:
       - content: |
 {{ .registryCACert | indent 10 }}
         owner: root:root
-        path: "/etc/containerd/certs.d/{{.registryMirrorConfiguration}}/ca.crt"
+        path: "/etc/containerd/certs.d/{{index .registryMirrorConfiguration "base"}}/ca.crt"
 {{- end }}
 {{- if .registryMirrorConfiguration }}
       - content: |
           [plugins."io.containerd.grpc.v1.cri".registry.mirrors]
             [plugins."io.containerd.grpc.v1.cri".registry.mirrors."public.ecr.aws"]
-              endpoint = ["https://{{.registryMirrorConfiguration}}"]
+              endpoint = ["https://{{index .registryMirrorConfiguration "public.ecr.aws"}}"]
```

Comment:
why don't just iterate over the mappings and them without checking for specific keys?